parsel/selector.py

```diff
@@ -39,8 +44,15 @@ def create_root_node(text, parser_cls, base_url=None):
     """Create root node for text using given parser class.
     """
     body = text.strip().replace('\x00', '').encode('utf8') or b'<html/>'
-    parser = parser_cls(recover=True, encoding='utf8')
```

Comment:
I think `parser_cls` made sense here when both classes had the same API. Now that we are introducing a new class that does not have the same signature, maybe we should switch to a different approach.

For example, instead of a class, we could use a function that returns a valid `root`, and modify `_ctgroup` so that `_parser` values are such functions.

That way, the code here remains independent of the `_parser` value.

What do you think?